src/cargo/util/errors.rs

```diff
@@ -1,101 +1,148 @@
 #![allow(unknown_lints)]
 
-use std::error::Error;
 use std::fmt;
-use std::io;
-use std::num;
 use std::process::{Output, ExitStatus};
 use std::str;
-use std::string;
 
 use core::TargetKind;
+use failure::{Context, Error, Fail};
 
-use curl;
-use git2;
-use semver;
-use serde_json;
-use toml;
-use registry;
-use ignore;
-
-error_chain! {
-    types {
-        CargoError, CargoErrorKind, CargoResultExt, CargoResult;
-    }
+pub use failure::Error as CargoError;
+pub type CargoResult<T> = Result<T, Error>;
 
-    links {
-        CrateRegistry(registry::Error, registry::ErrorKind);
-    }
+pub trait CargoResultExt<T, E> {
```

Comment:
Why still have this instead of using failure's ResultExt?

Reply:
Oh this is purely transitionary, I wanted to make sure we got past the test at least before blanket renaming `chain_err` to `with_context` everywhere. I plan on doing that change after this PR lands (just to cut down on merge conflicts hopefully)

Reply:
Sounds good!